src/engine/sync/enginesync.cpp

```diff
@@ -429,31 +429,46 @@ void EngineSync::deactivateSync(Syncable* pSyncable) {
     }
 }
 
-EngineChannel* EngineSync::pickNonSyncSyncTarget(EngineChannel* pDontPick) const {
-    EngineChannel* pFirstNonplayingDeck = nullptr;
+Syncable* EngineSync::pickNonSyncSyncTarget(EngineChannel* pDontPick) const {
+    // First choice: the sync master, if it's a deck
+    if (m_pMasterSyncable &&
+            m_pMasterSyncable->getChannel() &&
+            m_pMasterSyncable->getChannel() != pDontPick) {
+        return m_pMasterSyncable;
+    }
```

Comment:
An explicit master should be never synced to anything else. So we should return nullptr if pDontPick is an expicite master.

Reply:
it's not the job of this function to decide who should call it.  Instead, the calling function should check to see if it's an explicit master and shouldn't be looking for a sync source in the first place.  (Also, pickSyncTarget is also used by KeyControl, so this isn't only used for bpm matching).

Fixed by changing the call points of pickSyncTarget in bpmcontrol.